rclcpp/src/rclcpp/parameter_service.cpp

```diff
@@ -93,7 +85,7 @@ ParameterService::ParameterService(
         } catch (const rclcpp::exceptions::ParameterNotDeclaredException & ex) {
           RCLCPP_DEBUG(rclcpp::get_logger("rclcpp"), "Failed to set parameter: %s", ex.what());
           result.successful = false;
-          result.reason = ex.what();
+          result.reason = "Parameter is not declared by the node, which only allows declared parameters";
```

Comment:
i agree that this should be defined as constant in https://github.com/ros2/rcl_interfaces/blob/rolling/rcl_interfaces/msg/SetParametersResult.msg

Reply:
I've opened a PR for that (https://github.com/ros2/rcl_interfaces/pull/164) and updated this PR to depend on it.